src/radical/entk/utils/sync_initiator.py

```diff
@@ -39,25 +49,40 @@ def sync_with_master(obj, obj_type, channel, queue, logger, local_prof):
     reply_queue = sid + '-' + reply_queue
 
     while True:
-        #self._logger.info('waiting for ack')
+
+      # self._logger.info('waiting for ack')
 
         method_frame, props, body = channel.basic_get(queue=reply_queue)
 
         if body:
+
+            # AM: what happens to messages which don't match the `if` below
+            # - it seems like they are doscarded?  Why?
+
```

Comment:
Messages from RMQ can come out of order. Messages that are unacknowledged will
get resent, so no messages are discarded actually. We acknowledge a processed
message in line 82.

Thanks for noticing that. I will an inline comment regarding this.

Reply:
@vivek-bala ping?

Reply:
@vivek-bala: ping?